kube-runtime/src/wait.rs

```diff
@@ -237,6 +242,69 @@ pub mod conditions {
         }
     }
 
+    /// An await condition for `Deployment` that returns `true` once the latest deployment has completed
+    ///
+    /// This looks for the condition that Kubernetes sets for completed deployments:
+    /// <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment>
+    #[must_use]
+    pub fn is_deployment_completed() -> impl Condition<Deployment> {
+        |obj: Option<&Deployment>| {
+            if let Some(depl) = &obj {
+                if let Some(s) = &depl.status {
+                    if let Some(conds) = &s.conditions {
+                        if let Some(dcond) = conds.iter().find(|c| {
+                            c.type_ == "Progressing" && c.reason == Some("NewReplicaSetAvailable".to_string())
+                        }) {
+                            return dcond.status == "True";
+                        }
+                    }
+                }
+            }
+            false
+        }
+    }
+
+    /// An await condition for `Service`s of type `LoadBalancer` that returns `true` once the backing load balancer has an external IP or hostname
+    #[must_use]
+    pub fn is_service_loadbalancer_provisioned() -> impl Condition<Service> {
+        |obj: Option<&Service>| {
+            if let Some(svc) = &obj {
+                // ignore services that are not type LoadBalancer (return true immediately)
+                if let Some(spec) = &svc.spec {
+                    if spec.type_ != Some("LoadBalancer".to_string()) {
+                        return true;
+                    }
```

Comment:
i guess this is the only questionable thing, and it's not really a fault of yours, but more about us conflating bools and options. i'll think a bit about https://github.com/kube-rs/kube/pull/1498 after this pr.

Reply:
Yeah, I don't think there's any better way to handle this at the moment.